sound/soc/sof/intel/hda.c

```diff
@@ -635,6 +639,8 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 #endif
 	hda_codec_i915_exit(sdev);
 
+	hda_dsp_power_state_reset(hda);
```

Comment:
@keyonjie this looks incorrect. Typically when you remove the SOF device, you resume it and bring it back to full power. So the DSP cant be in D3. 

Besides reset is hardly intuitive to convey the message that the DSP is in D3. please change the name.

Reply:
@keyonjie I dont know in my opinion, the mutex isnt required, so just set the power state directly here and remove the init/reset functions